core/binding/src/store/primitive.rs

```diff
@@ -0,0 +1,205 @@
+use std::io::Cursor;
+use std::mem;
+
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use bytes::Bytes;
+use cita_trie::DB as TrieDB;
+
+use protocol::traits::{ServiceState, StoreBool, StoreString, StoreUint64};
+use protocol::types::Hash;
+use protocol::ProtocolResult;
+
+use crate::state::GeneralServiceState;
+use crate::store::StoreType;
+use crate::BindingError;
+
+pub struct DefaultStoreBool<DB: TrieDB> {
+    state: GeneralServiceState<DB>,
+    key:   Hash,
+}
+
+impl<DB: TrieDB> DefaultStoreBool<DB> {
+    pub fn new(state: GeneralServiceState<DB>, var_name: &str) -> Self {
+        Self {
+            state,
+            key: Hash::digest(Bytes::from(var_name.to_owned() + "bool")),
+        }
+    }
+}
+
+impl<DB: TrieDB> StoreBool for DefaultStoreBool<DB> {
+    fn get(&self) -> ProtocolResult<bool> {
+        let opt_bs: Option<Bytes> = self.state.get(&self.key)?;
+
+        match opt_bs {
+            Some(bs) => {
+                let mut rdr = Cursor::new(bs.to_vec());
+                let u = rdr.read_u8().unwrap();
```

Comment:
Don't use `unwrap`, if it never happens you can use `expect` with some description.